pkg/sql/schemachanger/scplan/internal/scstage/build.go

```diff
@@ -857,9 +857,15 @@ func (bc buildContext) makeDescriptorStates(cur, next *Stage) map[descpb.ID]*scp
 			state.RelevantStatements, stmtRank,
 		)
 	}
+	isPruned := cur.Phase == scop.PostCommitNonRevertiblePhase
 	for i, t := range bc.targetState.Targets {
 		descID := screl.GetDescID(t.Element())
 		state := ds[descID]
+		if isPruned && cur.After[i] == t.TargetStatus {
+			// Remove satisfied targets once the schema change is no longer
+			// revertible.
+			continue
+		}
```

Comment:
We could be more aggressive in removing targets for the schema change state, and from releasing descriptors from the schema change job. I'm not sure how necessary it is, though. I think previously I'd left "done" descriptors in the job on the assumption that in the non-revertible post-commit phase the job would finish soon anyway and this wouldn't matter, but this perhaps interferes with the cumulative backup tests. To be investigated.